src/mame/marion/ms68k.cpp

```diff
@@ -0,0 +1,319 @@
+	void floppy_drive_select(int drive);
+
+	// Printer
+	uint8_t printer_r(offs_t offset);
+	void printer_w(offs_t offset, uint8_t data);
+};
+
+/* Input ports */
+static INPUT_PORTS_START( ms68k )
+	PORT_START("ABORT")
+	PORT_BIT(0x01, IP_ACTIVE_HIGH, IPT_BUTTON1) PORT_NAME("NMI button") PORT_CHANGED_MEMBER(DEVICE_SELF, ms68k_state, nmi_button, 0)
+INPUT_PORTS_END
+
+
+void ms68k_state::ms68k(machine_config &config)
+{
+	M68000(config, m_maincpu, 16_MHz_XTAL / 2);
+	m_maincpu->set_addrmap(AS_PROGRAM, &ms68k_state::mem_map);
+
+	// Set up DUART.
+
+	MC68681(config, m_duart, 16_MHz_XTAL / 2);
+	// DUART Interrupt = 3
+	m_duart->irq_cb().set_inputline(m_maincpu, M68K_IRQ_3);
+
+	// Set up terminal RS-232.
+
+	RS232_PORT(config, m_terminal, default_rs232_devices, "terminal");
+	m_duart->a_tx_cb().set(m_terminal, FUNC(rs232_port_device::write_txd));
+	m_terminal->rxd_handler().set(m_duart, FUNC(mc68681_device::rx_a_w));
+	// DUART IP0 = CTS
+	m_terminal->cts_handler().set(m_duart, FUNC(mc68681_device::ip0_w));
+	// DUART OP0 = RTS
+	m_duart->outport_cb().append(m_terminal, FUNC(rs232_port_device::write_rts)).bit(0);
+
+	// Set up modem RS-232.
+
+	RS232_PORT(config, m_modem, default_rs232_devices, nullptr);
+	m_duart->b_tx_cb().set(m_modem, FUNC(rs232_port_device::write_txd));
+	m_modem->rxd_handler().set(m_duart, FUNC(mc68681_device::rx_b_w));
+	// DUART IP1 = CTS
+	m_modem->cts_handler().set(m_duart, FUNC(mc68681_device::ip1_w));
+	// DUART OP1 = RTS
+	m_duart->outport_cb().append(m_modem, FUNC(rs232_port_device::write_rts)).bit(1);
+
+	// Set up SCSI.
+
+	NSCSI_BUS(config, m_scsibus);
+	NSCSI_CONNECTOR(config, "scsibus:0", default_scsi_devices, "harddisk", false);
+	NSCSI_CONNECTOR(config, "scsibus:1", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:2", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:3", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:4", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:5", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:6", default_scsi_devices, nullptr, false);
+	NSCSI_CONNECTOR(config, "scsibus:7").option_set("ncr5380", NCR53C80)
+		.machine_config([this](device_t *device) {
+			// SCSI Interrupt = 5
+			downcast<ncr53c80_device &>(*device).irq_handler().set_inputline(m_maincpu, M68K_IRQ_5);
+		});
+
+	// Set up FDC.
+
+	WD1772(config, m_fdc, 16_MHz_XTAL / 2);
+	// FDC Interrupt = 4
+	m_fdc->intrq_wr_callback().set_inputline(m_maincpu, M68K_IRQ_3);
+	// DUART OP3 = Side 1 Select
+	m_duart->outport_cb().append(FUNC(ms68k_state::floppy_side_w)).bit(3);
+	// DUART OP4 = Drive Select 0
+	m_duart->outport_cb().append(FUNC(ms68k_state::floppy_drive0_w)).bit(4);
+	// DUART OP5 = Drive Select 1
+	m_duart->outport_cb().append(FUNC(ms68k_state::floppy_drive1_w)).bit(5);
+	// DUART OP6 = Double Density
+	m_duart->outport_cb().append(m_fdc, FUNC(wd1772_device::dden_w)).bit(6);
```

Comment:
Same here, please move the DUART output routing up with the DUART instantiation.